mail_template_multi_attachment/models/mail_template.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2021 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import base64
+
+from odoo import _, exceptions, fields, models
+from odoo.tools.safe_eval import safe_eval, time
+
+
+class MailTemplate(models.Model):
+    _inherit = "mail.template"
+
+    template_report_ids = fields.One2many(
+        comodel_name="mail.template.report",
+        inverse_name="mail_template_id",
+    )
+
+    # pylint: disable=redefined-outer-name
+    def generate_email(self, res_ids, fields=None):
+        """
+        Inherit to generate attachments.
+        Inspired from original mail.template,generate_email(...) from Odoo.
+        :param res_ids: int or list of int
+        :param fields:
+        :return: dict
+        """
+        self.ensure_one()
+        multi_mode = True
+        results = super(MailTemplate, self).generate_email(res_ids, fields=fields)
+        if not self.template_report_ids:
+            return results
+        if isinstance(res_ids, int):
+            multi_mode = False
+            results = {res_ids: results}
+        # Generate attachments (inspired from Odoo); Just add new attachments
+        # into 'attachments' key
```

Comment:
Could you check my comment here? 
https://github.com/OCA/social/pull/928#discussion_r1129128908
Still, not a blocker but nice to have, if you have time.

Reply:
We made the same changes in this PR by separating it as separate method @simahawk